### PR TITLE
Update Java-9-Resources.md

### PR DESCRIPTION
**Pull request by SimonVerhoeven**
Added an extra resource.
Whilst this resource is mostly about Java 8, the 'Additions in Java 9' chapter does offer a nice look forward + interesting for existing codebases.

### CODE REVIEW

**Comment by neomatrix369** [comment]
@SimonVerhoeven Thanks for the resource although not a Java 9 resource, the book is a Java 8 one

**Comment by SimonVerhoeven** [comment]
@neomatrix369 whilst indeed mostly a java 8 resource, I thought it might be a nice addition thanks to the `Additions in Java 9` chapter since it helps with the move forwards.

**Comment by neomatrix369** [comment]
@SimonVerhoeven nice one, do you mind adding a small comment in few words around the link to mention this Java 9 aspect of the book, and then the PR is good to go, I'll merge it for you.

Thanks again.

**Comment by SimonVerhoeven** [comment]
@neomatrix369 done, and you're welcome.

**Comment by neomatrix369** [comment]
@SimonVerhoeven Have you pushed the changes to the parent repo? ;)

File: Java-9-Resources.md [hunk 1]
```diff
@@ -121,6 +121,7 @@ Misc:
 * [Java 9 Programming By Example](https://www.amazon.co.uk/Java-Programming-Example-Peter-Verhas/dp/178646828X) (Packt; Peter Verhas) - available on [Safari Books online](https://www.safaribooksonline.com/library/view/java-9-programming/9781786468284/)
 * [Java 9 Revealed](http://www.springer.com/gb/book/9781484225912?wt_mc=PPC.Google%20AdWords.3.EPR436.GoogleShopping_Product_UK&gclid=EAIaIQobChMItLHMoK3e1QIVSLHtCh0v8QnCEAQYASABEgIEpfD_BwE) (Springer; Kishori Sharan)
 * [Java 8 & 9 in Action, 2nd Edition](https://www.manning.com/books/java-8-and-9-in-action) (Manning; Raoul-Gabriel Urma, Mario Fusco, Alan Mycroft)
+* [Modern Java recipes](https://www.safaribooksonline.com/library/view/modern-java-recipes/9781491973165/) (O'Reilly Media, Inc.; Ken Kouson)
```
Comment (neomatrix369) [inline Java-9-Resources.md]: @SimonVerhoeven I was suggesting something like this:
* [Modern Java recipes](https://www.safaribooksonline.com/library/view/modern-java-recipes/9781491973165/ - Java 9 Additions chapter to help migrate from 8 to 9) (O'Reilly Media, Inc.; Ken Kouson)
` Java 9 Additions chapter to help migrate from 8 to 9` - you can write to the way you want it as long as it conveys the message

**Comment by neomatrix369** [comment]
@SimonVerhoeven Can we do the same one for the `exercise-solutions` branch as well pls

**Pull request merged**